includes/model/class-post.php

```diff
@@ -226,18 +229,20 @@ public function generate_object_type() {
 		return $object_type;
 	}
 
+	/**
+	 * Generates the content for the activitypub item.
+	 *
+	 * @return string the content
+	 */
 	public function generate_the_content() {
 		$post = $this->post;
 		$content = $this->get_post_content_template();
 
-		$content = \str_replace( '%title%', \get_the_title( $post->ID ), $content );
-		$content = \str_replace( '%excerpt%', $this->get_the_post_excerpt(), $content );
-		$content = \str_replace( '%content%', $this->get_the_post_content(), $content );
-		$content = \str_replace( '%permalink%', $this->get_the_post_link( 'permalink' ), $content );
-		$content = \str_replace( '%shortlink%', $this->get_the_post_link( 'shortlink' ), $content );
-		$content = \str_replace( '%hashtags%', $this->get_the_post_hashtags(), $content );
-		// backwards compatibility
-		$content = \str_replace( '%tags%', $this->get_the_post_hashtags(), $content );
+		// Register the shortcodes.
+		$shortcodes = new \Activitypub\Shortcodes( $post );
```

Comment:
I prefer using:

```
setup_postdata( $post );
$content = do_shortcode( $content );
wp_reset_postdata();
```

It makes the handling of the shortcodes easier and more generic (and more compatible with the loop).

Reply:
I'm not a fan of this technique, but I've coverted the code over to.  Adding the shortcodes even when not used isn't that big of a hit for overhead 🤷